Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py

```diff
@@ -0,0 +1,209 @@
+EMAIL_HTML_FIELD = 'emailbodyhtml'
+FROM_FIELD = 'emailfrom'
+FROM_DOMAIN_FIELD = 'fromdomain'
+MERGED_TEXT_FIELD = 'mereged_text'
+MIN_TEXT_LENGTH = 50
+DEFAULT_ARGS = {
+    'limit': '1000',
+    'incidentTypes': 'Phishing',
+    'exsitingIncidentsLookback': '100 days ago',
+}
+FROM_POLICY_TEXT_ONLY = 'TextOnly'
+FROM_POLICY_EXACT = 'Exact'
+FROM_POLICY_DOMAIN = 'Domain'
+
+FROM_POLICY = FROM_POLICY_TEXT_ONLY
+
+
+def get_existing_incidents(input_args):
+    global DEFAULT_ARGS
+    get_incidents_args = {}
+    for arg in ['incidentTypes', 'query', 'limit']:
+        if arg in input_args:
+            get_incidents_args[arg] = input_args[arg]
+        elif arg in DEFAULT_ARGS:
+            get_incidents_args[arg] = DEFAULT_ARGS[arg]
+    if 'exsitingIncidentsLookback' in input_args:
+        get_incidents_args['fromDate'] = input_args['exsitingIncidentsLookback']
+    elif 'exsitingIncidentsLookback' in DEFAULT_ARGS:
+        get_incidents_args['fromDate'] = DEFAULT_ARGS['exsitingIncidentsLookback']
+    incidents_query_res = demisto.executeCommand('GetIncidentsByQuery', get_incidents_args)
+    if is_error(incidents_query_res):
+        return_error(get_error(incidents_query_res))
+    incidents = json.loads(incidents_query_res[-1]['Contents'])
+    return incidents
+
+
+def extract_domain(address):
+    if address == '':
+        return ''
+    email_address = parseaddr(address)[1]
+    ext = NO_FETCH_EXTRACT(email_address)
+    return ext.domain
+
+
+def get_text_from_html(html):
+    # todo: change to docker which supports
+    soup = BeautifulSoup(html)
+    # kill all script and style elements
+    for script in soup(["script", "style"]):
+        script.extract()  # rip it out
+    # get text
+    text = soup.get_text()
+    # break into lines and remove leading and trailing space on each
+    lines = (line.strip() for line in text.splitlines())
+    # break multi-headlines into a line each
+    chunks = (phrase.strip() for line in lines for phrase in line.split("  "))
+    # drop blank lines
+    text = '\n'.join(chunk for chunk in chunks if chunk)
+    return text
+
+
+def preprocess_text_fields(incident):
+    email_body = email_subject = email_html = ''
+    if EMAIL_BODY_FIELD in incident:
+        email_body = incident[EMAIL_BODY_FIELD]
+    if EMAIL_HTML_FIELD in incident:
+        email_html = incident[EMAIL_HTML_FIELD]
+    if EMAIL_SUBJECT_FIELD in incident:
+        email_subject = incident[EMAIL_SUBJECT_FIELD]
+    if isinstance(email_html, float):
+        email_html = ''
+    if email_body is None or isinstance(email_body, float) or email_body.strip() == '':
+        email_body = get_text_from_html(email_html)
+    if isinstance(email_subject, float):
+        email_subject = ''
+    return email_subject + ' ' + email_body
+
+
+def preprocess_incidents_df(existing_incidents):
+    global MERGED_TEXT_FIELD, FROM_FIELD, FROM_DOMAIN_FIELD
+    incidents_df = pd.DataFrame(existing_incidents)
```

Comment:
because you are using "GetIncidentsByQuery" I think the incident you get already "flat" the CustomFields, so you can just drop this column (make sure it's true)

Reply:
It's true for the output of `GetIncidentsByQuery`, but it's not true for the new created incident.  Wanted to have a single preprocessing function, will make it more efficient